integrate file commenting into partner reporting script

pwnage101 opened the pull request:
* Also, update the _log docstring to not mislead that we are not
  currently sending logs to splunk (we are).

pwnage101 commented on tubular/scripts/retirement_partner_report.py:
```diff
@@ -45,6 +45,8 @@
 ERR_DRIVE_LISTING = -12
 
+NOTIFICATION_MESSAGE = "FIXME"
+
```
Gonna run this by docs for wording.  Also, taking recommendations.

pwnage101 replied:
ugh, why would I even waste their time.  I'm just going to use "This CSV report contains a list of edx.org usernames for learners who were registered in one or more of your organization's courses and have requested account retirement."

pwnage101 replied:
okay, I'm running it by docs anyway because of a gut feeling.

bmedx commented on tubular/scripts/retirement_partner_report.py:
```diff
@@ -244,16 +252,36 @@ def _push_files_to_google(config, partner_filenames):
+        _log('Attempting to add notification comments to uploaded csv files.')
+        drive.create_comments_for_files(file_ids, NOTIFICATION_MESSAGE)
+    except Exception as exc:  # pylint: disable=broad-except
+        # do not fail the script here, since comment errors are non-critical
```
👍

bmedx commented on tubular/scripts/retirement_partner_report.py:
```diff
@@ -244,16 +252,36 @@ def _push_files_to_google(config, partner_filenames):
+                file_id = drive.create_file_in_folder(folder_id, partner_filenames[partner], f, "text/csv")
             except Exception as exc:  # pylint: disable=broad-except
                 _fail_exception(ERR_DRIVE_UPLOAD, exc, 'Drive upload failed for: {}'.format(partner_filenames[partner]))
+        file_ids[partner] = file_id
```
Why a dict mapping here? It looks like only the values actually get used.

pwnage101 replied:
It just makes the interface sensible, but since this isn't a consumable library function I'll just simplify it to return a flat list.

bmedx commented on tubular/scripts/retirement_partner_report.py:
```diff
@@ -45,6 +45,11 @@
 ERR_CLEANUP = -11
 ERR_DRIVE_LISTING = -12
 
+NOTIFICATION_MESSAGE = \
```
I'm sure this message will be the cause of much discussion. My $.02 is that whoever is seeing it is likely seeing it every week, so I'd go for something that's more "Your learner retirement report from edX has been published to Drive" over explaining what the file is.

pwnage101 commented:
Additional changes:

* report_file_ids flattened to a list (was a mapping where the keys were ignored)
* added a `--comments` (default) and complementary `--no_comments` flag.
* modified reporting tests to also check that `create_comments_for_files` is being called correctly
* updated a few incorrect usages of `assert_called_once()` (was `assert mock_object.called_once()` which would unconditionally succeed)

bmedx commented on tubular/google_api.py:
```diff
@@ -249,3 +243,37 @@ def callback(request_id, response, exception):  # pylint: disable=unused-argumen
+            content (str): content/message of the comment for every file.
+            fields (str): comma separated list of fields to describe each comment resource in the response.
+
+        Returns: dict mapping of file_id to comment resource (dict).  The contents of the comment resources is dictated
```
Tiniest of nits: "are dictated"

doctoryes commented on tubular/scripts/retirement_partner_report.py:
```diff
@@ -56,8 +59,8 @@ class PartnerDoesNotExist(Exception):
 
 def _log(info):
```
My PR here: https://github.com/edx/tubular/pull/250 removes this method altogether.

pwnage101 replied:
ah, well then I'll let you deal with the rebase!
